src/background/utils.ts

```diff
@@ -81,11 +81,37 @@ export const getTabId = (sender: Runtime.MessageSender): number => {
   return notNullOrUndef(notNullOrUndef(sender.tab, 'sender.tab').id, 'tab.id')
 }
 
+export const getTab = (sender: Runtime.MessageSender): Tabs.Tab => {
+  return notNullOrUndef(notNullOrUndef(sender.tab, 'sender.tab'), 'tab')
+}
+
 export const getSender = (sender: Runtime.MessageSender) => {
   const tabId = getTabId(sender)
   const frameId = notNullOrUndef(sender.frameId, 'sender.frameId')
-  return { tabId, frameId }
+  const tab = getTab(sender)
+
+  return { tabId, frameId, url: sender.url, tab }
 }
 
 export const computeRate = (rate: string, sessionsCount: number) =>
   (+rate / sessionsCount).toString()
+
+export const getHash = async (str: string) => {
```

Comment:
We can remove this function as unused, or feel free to keep it. I'd recommend passing algorithm and format to make it more usable (can use in `OpenPayments.verifyInteractionHash`).. or maybe something for future only..

```ts
export const getHash = async (str: string, algorithm: AlgorithmIdentifier, format: 'hex' | 'base64') => {
```